src/optimum/rbln/diffusers/pipelines/cosmos/cosmos_guardrail.py

```
@@ -0,0 +1,492 @@
+        rbln_files = list(model_save_dir.glob("*.rbln"))
+        rbln_config_filenames = list(model_save_dir.glob("rbln_config.json"))
+        validate_files(rbln_files, rbln_config_filenames, f"directory {model_save_dir}")
+
+        from_export_method = isinstance(rbln_config, RBLNModelConfig) and rbln_compiled_models is not None
+        if not from_export_method:
+            rbln_config_as_kwargs = {f"rbln_{key}": value for key, value in rbln_config.items()}
+            rbln_config, kwargs = RBLNAutoConfig.load(
+                model_save_dir, passed_rbln_config=None, kwargs=rbln_config_as_kwargs, return_unused_kwargs=True
+            )
+
+        compiled_models = Path(model_save_dir).glob("*.rbln")
+        rbln_compiled_models = {cm.stem: rebel.RBLNCompiledModel(cm) for cm in compiled_models}
+        return cls._from_compiled_models(rbln_compiled_models, rbln_config, model_path=model_path, subfolder=subfolder)
+
+    @classmethod
+    def _from_compiled_models(
+        cls,
+        rbln_compiled_models: Dict[str, rebel.RBLNCompiledModel],
+        rbln_config: "RBLNModelConfig",
+        model_save_dir: Optional[Union[str, Path, TemporaryDirectory]] = None,
+        subfolder="",
+        **kwargs,
+    ):
+        compiled_model_names = [cfg.compiled_model_name for cfg in rbln_config.compile_cfgs]
+        rbln_compiled_models = [rbln_compiled_models[cm_name] for cm_name in compiled_model_names]
+
+        # create runtimes only if `rbln_create_runtimes` is enabled
+        try:
+            models = (
+                cls._create_runtimes(rbln_compiled_models, rbln_config)
+                if rbln_config.create_runtimes
+                else UnavailableRuntime()
+            )
+
+        except rebel.core.exception.RBLNRuntimeError as e:
+            logger.warning(
+                f"Failed to create the runtime for the model due to a runtime error: {e.__class__.__name__} - {e}"
+            )
+            models = UnavailableRuntime()
```

Please fix this as main branch do. 

exception should be raised(`main`) instead of skipping (`this branch`).